core/src/test/java/org/apache/iceberg/TableTestBase.java

```diff
@@ -410,11 +410,19 @@ void validateSnapshot(Snapshot old, Snapshot snap, Long sequenceNumber, DataFile
       if (sequenceNumber != null) {
         V1Assert.assertEquals(
             "Data sequence number should default to 0", 0, entry.dataSequenceNumber().longValue());
+        V1Assert.assertEquals(
+            "Data sequence number should default to 0",
```

Comment:
do we need assert on file sequence number too?

Reply:
@gaborkaszab, you are correct that rewrites may preserve the initial data sequence number. This method would not work for such rewrites. I believe we have other tests that pass an iterator of sequence numbers that should be used in those cases.